### ISSUE TITLE
Ahem -- help!!

### ISSUE THREAD

**Issue by briantbutton** [drdrew42/renderer#87]
drdrew42, thank you for creating this repo!  It is super helpful to look at a smaller cut of the Webwork functionality.

We are running an Ubuntu server.  We need to convert problem text string to an HTML text string as a utility function for this server.  The simple thing would be to just call a perl function with some PGML (+ other parameters) and get back the response. 

It seemed like it would be a simple task to write a module to call several methods in Problem.pm serially, eventually getting back the result.  That is not working, simply because I know so freakin' little Perl script.  (Maybe other reasons too.)  I am not setting up the parameters right.

If you have any suggestions, I will eagerly take them.  Even a few lines of untested code that are in the right ballpark would advance the cause.

There is always the fallback of running the Webwork renderer against a port, as this repo instructs, and connecting our existing web server to it.  Pipe the conversion dialog through http.

**Comment by drdrew42** [comment]
@briantbutton -- I'm kind of in the dark about what it is that you're aiming to do. The purpose of the renderer project is to separate out the rendering process of pg files from the rest of the WW2 codebase. You say that you're looking to convert 'problem text string' to 'HTML text string', which is (generally) the purpose of this project. What is it that isn't working for you? Are you using pg files from the OPL or do you have your own content that you're trying to render?

**Comment by briantbutton** [comment]
Hey, yeah, its own content.  In fact it is interactive exercises embedded in textbooks found here.  (https://yoshiwarabooks.org/)   We build local servers that carry content anywhere in the world, and specifically, places that don't have internet.  They hold scads of educational materials, Wikipedia, Khan Academy lectures, PhET simulations.
![angle-closed](https://user-images.githubusercontent.com/16803497/165656769-ea030b5b-764c-449f-a4d8-d706c51e9080.jpeg)

The clients connect with the webserver on our local box.  They can edit collaborative documents, download videos, etc.  We do proxy to a Wikipedia server, also running in our box.  (See Kiwix.)  A bit painful but we got Wikipedia.  No more of those.

Skipping over details, I would say it seemed that any way of doing it using Webwork code, even with your head start was gonna be too trying, given the other things on our plate.  So, for now, we will ship without the interactive exercises.  And we closed the issue.

Thank you for noticing.

A BIT MORE
I believe the world could use a simple interactive quiz utility, that would be an expansion on html.  It could be a plug-in for ePubs and a open-source JavaScript utility for web apps.  The authoring language should be web appropriate for people from all disciplines.  (Hide or drop Perl script.)  And, for our purposes, it *must* run in the browser.  <Bring the soap box>

MY SOAPBOX
Technologists design as if the internet was free, fast and reliable everywhere.  Most Africans pay heinous usage fees if they run over a data cap.  Latin America internet coverage is mostly from a relatively tenuous RF network.  It works most of the time but it would be so much better if they could study when the net is down.  Children in India, Indonesia and many other countries have similar problems.

AND SO
That's probably more than what you were looking for.  I appreciate the dialog.
